arbnode/transaction_streamer.go

```diff
@@ -754,6 +785,36 @@ func (s *TransactionStreamer) countDuplicateMessages(
 					}
 					dbMessageParsed.Message.BatchGasCost = batchGasCostBkup
 				}
+				msg := nextMessage.MessageWithMeta.Message
+				if arbos.IsEspressoMsg(dbMessageParsed.Message) &&
+					arbos.IsEspressoMsg(msg) &&
+					!bytes.Equal(msg.L2msg, dbMessageParsed.Message.L2msg) {
```

Comment:
@ImJeremyHe can you remind me why we do this. And what seems a bit unusual is that this happens in the `countDuplicateMessages` function. So it would be great to have a commend explaining why.